clients/pkg/promtail/targets/file/tailer.go

```diff
@@ -217,6 +236,24 @@ func (t *tailer) isRunning() bool {
 	return t.running.Load()
 }
 
+func (t *tailer) convertToUTF8(text string) (string, error) {
+	level.Debug(t.logger).Log("msg", "Converting log encoding", "from", t.encoding, "to", "UTF8")
+
+	encoding, err := ianaindex.IANA.Encoding(t.encoding)
+	if err != nil {
+		return "", errors.Wrap(err, "error doing IANA encoding")
+	}
+
+	utf8Reader := transform.NewReader(strings.NewReader(text), encoding.NewDecoder())
```

Comment:
Should you use https://pkg.go.dev/golang.org/x/text/transform#String instead ? Seem more fitted for your usage.

Reply:
good call, thanks